src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,42 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                for stepName in task.listAllStepNames():
+                    sh = task.getStepHelper(stepName)
+                    origCores = max(origCores, sh.getNumberOfCores())
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    origMaxRSS = int(origMaxRSS/1000.) # HTCondor expects MB; we get KB.
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # HTCondor doesn't explicitly scale VSize; it's also not clear what
+                # resources this manages (as we already watch the memory use) or how
+                # it should relate to other resources (such as memory or cores used).
+                # Hence, we simply remove it if we change anything about the memory.
+                if origMaxRSS and (resources['memory'] != origMaxRSS) and ('maxVSize' in args):
+                    del args['maxVSize']
+                # Scale our timeouts according to the scaling of the cores.  *NOTE*
+                # the job ad contains the real scaling here - unfortunately, it uses
+                # ClassAd expressions and we can only parse values.  Something for the
+                # future.
+                # As the scaling is not perfectly linear (efficiency is closer to 95%),
+                # we add a somewhat-arbitrarily-chosen 60 minutes (timeouts are in seconds).
+                coreScaling = resources['cores'] / origCores
+                if 'softTimeout' in args:
+                    args['softTimeout'] = int(args['softTimeout']/coreScaling) + 3600
+                if 'hardTimeout' in args:
+                    args['hardTimeout'] = int(args['hardTimeout']/coreScaling) + 3600
+                # If we did base maxRSS off the memory in the HTCondor slot, subtract a bit
+                # off the top so watchdog triggers before HTCondor does.
+                if origMaxRSS and (resources['memory'] != origMaxRSS):
+                    args['maxRSS'] = 1000*(resources['memory'] - 50) #Convert back to KB
```

Comment:
In order to have a better job tracking (and future debugging), how about we change `resources['memory']` from the machine memory allocated to the exact `RequestMemory` value?

On the watchdog side, it makes more sense to kill a job only after it goes *beyond* the RequestMemory, so I'd get rid of the `-50`, maybe we add 50 actually.